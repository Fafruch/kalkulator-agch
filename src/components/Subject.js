/**
 * Created by Fafruch on 17/07/2017.
 */

import React from 'react'
import PropTypes from 'prop-types'
import ToggleSubject from './ToggleSubject'
import DeleteSubject from './DeleteSubject'
import PrimarySubjectsDatalist from './PrimarySubjectsDatalist'
import LingualSubjectsDatalist from './LingualSubjectsDatalist'
import { primarySubjects } from '../constants/SubjectTypes'

let Subject = ({ id, name, elementaryScore, advancedScore,
                 active, onChange, onToggleClick, onDelete, subjectsType }) => {
  let formInput = {
    name,
    elementaryScore,
    advancedScore
  }

  return (
    <table className='table'>
      <tbody>
        <tr>
          <th>ID</th>
          <th>Przedmiot</th>
          <th>Wynik z matury podstawowej</th>
          <th>Wynik z matury rozszerzonej</th>
          <th>{/* {active ? 'Nie bierz pod uwagę': 'Bierz pod uwagę'} */}</th>
          <th>{/* Usuń {subjectsType === primarySubjects ? 'przedmiot' : 'język'} */}</th>
        </tr>
        <tr>
          <td>
            {id}
          </td>
          <td>
            <input
              type='text'
              list={subjectsType}
              ref={node => { formInput.name = node }}
              onChange={() => onChange(id, formInput.name.value, elementaryScore, advancedScore)}
              className={active ? 'subject-active' : 'subject-not-active'}
            />

            {subjectsType === primarySubjects
              ? <PrimarySubjectsDatalist subjectsType={subjectsType} />
              : <LingualSubjectsDatalist subjectsType={subjectsType} />
            }

          </td>
          <td>
            <input
              type='range'
              defaultValue='0'
              min='0'
              max='100'
              ref={node => { formInput.elementaryScore = node }}
              onChange={() => onChange(id, name, parseInt(formInput.elementaryScore.value, 10), advancedScore)}
            />

            &nbsp; {elementaryScore === 0 ? 'Brak' : elementaryScore + '%'}

            {/* // trzeba polaczyc jakos, zeby dzialalo
            <input
              type='text'
              value={elementaryScore}
              ref={node => { formInput.elementaryScore = node }}
              onChange={() => onChange(id, name, type, parseInt(formInput.elementaryScore.value, 10), advancedScore)}
              style={{width: 20}}
            >
            </input>
            */}
          </td>
          <td>
            <input
              type='range'
              defaultValue='0'
              min='0'
              max='100'
              ref={node => { formInput.advancedScore = node }}
              onChange={() => onChange(id, name, elementaryScore, parseInt(formInput.advancedScore.value, 10))}
            />

            &nbsp; {advancedScore === 0 ? 'Brak' : advancedScore + '%'}

            {/* // tu tez
            <input
              type='text'
              value={advancedScore}
              ref={node => { formInput.elementaryScore = node }}
              onChange={() => onChange(id, name, type, elementaryScore, parseInt(formInput.advancedScore.value, 10))}
              style={{width: 20}}
            >
            </input>
            */}
          </td>
          <td>
            <ToggleSubject
              id={id}
              active={active}
              onToggleClick={onToggleClick}
            />
          </td>
          <td>
            <DeleteSubject
              id={id}
              onDelete={onDelete}
            />
          </td>
        </tr>
      </tbody>
    </table>
  )
}

Subject.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  elementaryScore: PropTypes.number.isRequired,
  advancedScore: PropTypes.number.isRequired,
  active: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
  onToggleClick: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired,
  subjectsType: PropTypes.string.isRequired
}

export default Subject
