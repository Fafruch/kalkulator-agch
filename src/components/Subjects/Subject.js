import React from 'react'
import PropTypes from 'prop-types'

import ToggleSubject from './ToggleSubject'
import DeleteSubject from './DeleteSubject'
import PrimarySubjectsDatalist from './PrimarySubjectsDatalist'
import LingualSubjectsDatalist from './LingualSubjectsDatalist'
import { PRIMARY_SUBJECTS } from '../../constants/SubjectTypes'

let Subject = ({ id, name, elementaryScore, advancedScore, active,
                 onChange, onToggleClick, onDelete, subjectsType, iterator }) => {
  let formInput = {
    name,
    elementaryScore,
    advancedScore
  }

  return (
    <table className='table mb-0'>
      <tbody>
        <tr>
          <th>ID</th>
          <th>Przedmiot</th>
          <th>Wynik z matury podstawowej</th>
          <th>Wynik z matury rozszerzonej</th>
          <th>{/* {active ? 'Nie bierz pod uwagę': 'Bierz pod uwagę'} */}</th>
          <th>{/* Usuń {subjectsType === PRIMARY_SUBJECTS ? 'przedmiot' : 'język'} */}</th>
        </tr>
        <tr>
          <td>
            {iterator}
          </td>
          <td>
            <input
              type='text'
              list={subjectsType}
              ref={node => { formInput.name = node }}
              defaultValue={name}
              onChange={() => onChange(id, formInput.name.value, elementaryScore, advancedScore)}
              className={active ? 'subject-active subject-input-field' : 'subject-not-active subject-input-field'}
            />

            {subjectsType === PRIMARY_SUBJECTS
              ? <PrimarySubjectsDatalist subjectsType={subjectsType} />
              : <LingualSubjectsDatalist subjectsType={subjectsType} />
            }
          </td>
          <td>
            <input
              type='range'
              defaultValue={elementaryScore}
              min='0'
              max='100'
              ref={node => { formInput.elementaryScore = node }}
              onChange={() => onChange(id, name, parseInt(formInput.elementaryScore.value, 10), advancedScore)}
            />

            &nbsp; {elementaryScore === 0 ? 'Brak' : elementaryScore + '%'}
          </td>
          <td>
            <input
              type='range'
              defaultValue={advancedScore}
              min='0'
              max='100'
              ref={node => { formInput.advancedScore = node }}
              onChange={() => onChange(id, name, elementaryScore, parseInt(formInput.advancedScore.value, 10))}
            />

            &nbsp; {advancedScore === 0 ? 'Brak' : advancedScore + '%'}
          </td>
          <td>
            <ToggleSubject
              id={id}
              active={active}
              onToggleClick={onToggleClick}
            />
          </td>
          <td>
            <DeleteSubject
              id={id}
              onDelete={onDelete}
            />
          </td>
        </tr>
      </tbody>
    </table>
  )
}

Subject.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  elementaryScore: PropTypes.number.isRequired,
  advancedScore: PropTypes.number.isRequired,
  active: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
  onToggleClick: PropTypes.func.isRequired,
  onDelete: PropTypes.func.isRequired,
  subjectsType: PropTypes.string.isRequired,
  iterator: PropTypes.number.isRequired
}

export default Subject
