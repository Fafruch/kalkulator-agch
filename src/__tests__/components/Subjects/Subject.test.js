import React from 'react'
import { shallow } from 'enzyme'

import Subject from '../../../components/Subjects/Subject'

const setup = (setupProps = {}) => {
  const defaultProps = {
    subject: {
      id: '0',
      name: 'Matematyka',
      elementaryScore: 100,
      advancedScore: 82,
      maxScore: {
        value: 182,
        formula: 'A/B'
      },
      active: true
    },
    subjectsType: 'primary',
    subjectNumber: 2,
    onChange: jest.fn(),
    onToggleClick: jest.fn(),
    onDelete: jest.fn()
  }
  const props = { ...defaultProps, ...setupProps }
  const wrapper = shallow(<Subject {...props} />)

  return {
    props,
    wrapper
  }
}

describe('Subject component', () => {
  test('should render', () => {
    const { wrapper } = setup()

    expect(wrapper).toMatchSnapshot()
  })
  test('should render % when the score is not 0', () => {
    let { wrapper } = setup()

    expect(wrapper.find('td').at(2).text()).toContain('%')

    ;({ wrapper } = setup({
      subject: {
        id: '0',
        name: 'Matematyka',
        elementaryScore: 0,
        advancedScore: 82,
        maxScore: {
          value: 182,
          formula: 'A/B'
        },
        active: true
      }
    }))

    expect(wrapper.find('td').at(2).text()).not.toContain('%')
  })
})
