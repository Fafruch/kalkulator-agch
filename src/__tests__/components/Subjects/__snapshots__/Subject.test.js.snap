// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subject component should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <table
    className="table mb-0"
>
    <tbody>
        <tr>
            <th>
                ID
            </th>
            <th>
                Przedmiot
            </th>
            <th>
                Wynik z matury podstawowej
            </th>
            <th>
                Wynik z matury rozszerzonej
            </th>
            <th />
            <th />
        </tr>
        <tr>
            <td>
                2
            </td>
            <td>
                <NameInput
                    onChange={[Function]}
                    subject={
                        Object {
                            "active": true,
                            "advancedScore": 82,
                            "elementaryScore": 100,
                            "id": "0",
                            "maxScore": Object {
                              "formula": "A/B",
                              "value": 182,
                            },
                            "name": "Matematyka",
                          }
                    }
                    subjectsType="primary"
                />
            </td>
            <td>
                <ScoreRangeInput
                    examType="elementary"
                    onChange={[Function]}
                    subject={
                        Object {
                            "active": true,
                            "advancedScore": 82,
                            "elementaryScore": 100,
                            "id": "0",
                            "maxScore": Object {
                              "formula": "A/B",
                              "value": 182,
                            },
                            "name": "Matematyka",
                          }
                    }
                    subjectsType="primary"
                />
                <ScoreTextInput
                    examType="elementary"
                    onChange={[Function]}
                    subject={
                        Object {
                            "active": true,
                            "advancedScore": 82,
                            "elementaryScore": 100,
                            "id": "0",
                            "maxScore": Object {
                              "formula": "A/B",
                              "value": 182,
                            },
                            "name": "Matematyka",
                          }
                    }
                    subjectsType="primary"
                />
                %
            </td>
            <td>
                <ScoreRangeInput
                    examType="advanced"
                    onChange={[Function]}
                    subject={
                        Object {
                            "active": true,
                            "advancedScore": 82,
                            "elementaryScore": 100,
                            "id": "0",
                            "maxScore": Object {
                              "formula": "A/B",
                              "value": 182,
                            },
                            "name": "Matematyka",
                          }
                    }
                    subjectsType="primary"
                />
                <ScoreTextInput
                    examType="advanced"
                    onChange={[Function]}
                    subject={
                        Object {
                            "active": true,
                            "advancedScore": 82,
                            "elementaryScore": 100,
                            "id": "0",
                            "maxScore": Object {
                              "formula": "A/B",
                              "value": 182,
                            },
                            "name": "Matematyka",
                          }
                    }
                    subjectsType="primary"
                />
                %
            </td>
            <td>
                <ToggleSubject
                    active={true}
                    id="0"
                    onToggleClick={[Function]}
                    subjectsType="primary"
                />
            </td>
            <td>
                <DeleteSubject
                    id="0"
                    onDelete={[Function]}
                    subjectsType="primary"
                />
            </td>
        </tr>
    </tbody>
</table>,
  "nodes": Array [
    <table
      className="table mb-0"
>
      <tbody>
            <tr>
                  <th>
                        ID
                  </th>
                  <th>
                        Przedmiot
                  </th>
                  <th>
                        Wynik z matury podstawowej
                  </th>
                  <th>
                        Wynik z matury rozszerzonej
                  </th>
                  <th />
                  <th />
            </tr>
            <tr>
                  <td>
                        2
                  </td>
                  <td>
                        <NameInput
                              onChange={[Function]}
                              subject={
                                    Object {
                                          "active": true,
                                          "advancedScore": 82,
                                          "elementaryScore": 100,
                                          "id": "0",
                                          "maxScore": Object {
                                            "formula": "A/B",
                                            "value": 182,
                                          },
                                          "name": "Matematyka",
                                        }
                              }
                              subjectsType="primary"
                        />
                  </td>
                  <td>
                        <ScoreRangeInput
                              examType="elementary"
                              onChange={[Function]}
                              subject={
                                    Object {
                                          "active": true,
                                          "advancedScore": 82,
                                          "elementaryScore": 100,
                                          "id": "0",
                                          "maxScore": Object {
                                            "formula": "A/B",
                                            "value": 182,
                                          },
                                          "name": "Matematyka",
                                        }
                              }
                              subjectsType="primary"
                        />
                        <ScoreTextInput
                              examType="elementary"
                              onChange={[Function]}
                              subject={
                                    Object {
                                          "active": true,
                                          "advancedScore": 82,
                                          "elementaryScore": 100,
                                          "id": "0",
                                          "maxScore": Object {
                                            "formula": "A/B",
                                            "value": 182,
                                          },
                                          "name": "Matematyka",
                                        }
                              }
                              subjectsType="primary"
                        />
                        %
                  </td>
                  <td>
                        <ScoreRangeInput
                              examType="advanced"
                              onChange={[Function]}
                              subject={
                                    Object {
                                          "active": true,
                                          "advancedScore": 82,
                                          "elementaryScore": 100,
                                          "id": "0",
                                          "maxScore": Object {
                                            "formula": "A/B",
                                            "value": 182,
                                          },
                                          "name": "Matematyka",
                                        }
                              }
                              subjectsType="primary"
                        />
                        <ScoreTextInput
                              examType="advanced"
                              onChange={[Function]}
                              subject={
                                    Object {
                                          "active": true,
                                          "advancedScore": 82,
                                          "elementaryScore": 100,
                                          "id": "0",
                                          "maxScore": Object {
                                            "formula": "A/B",
                                            "value": 182,
                                          },
                                          "name": "Matematyka",
                                        }
                              }
                              subjectsType="primary"
                        />
                        %
                  </td>
                  <td>
                        <ToggleSubject
                              active={true}
                              id="0"
                              onToggleClick={[Function]}
                              subjectsType="primary"
                        />
                  </td>
                  <td>
                        <DeleteSubject
                              id="0"
                              onDelete={[Function]}
                              subjectsType="primary"
                        />
                  </td>
            </tr>
      </tbody>
</table>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Subject
        onChange={[Function]}
        onDelete={[Function]}
        onToggleClick={[Function]}
        subject={
                Object {
                        "active": true,
                        "advancedScore": 82,
                        "elementaryScore": 100,
                        "id": "0",
                        "maxScore": Object {
                          "formula": "A/B",
                          "value": 182,
                        },
                        "name": "Matematyka",
                      }
        }
        subjectNumber={2}
        subjectsType="primary"
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "onChange": [Function],
          "onDelete": [Function],
          "onToggleClick": [Function],
          "subject": Object {
            "active": true,
            "advancedScore": 82,
            "elementaryScore": 100,
            "id": "0",
            "maxScore": Object {
              "formula": "A/B",
              "value": 182,
            },
            "name": "Matematyka",
          },
          "subjectNumber": 2,
          "subjectsType": "primary",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <table
          className="table mb-0"
>
          <tbody>
                    <tr>
                              <th>
                                        ID
                              </th>
                              <th>
                                        Przedmiot
                              </th>
                              <th>
                                        Wynik z matury podstawowej
                              </th>
                              <th>
                                        Wynik z matury rozszerzonej
                              </th>
                              <th />
                              <th />
                    </tr>
                    <tr>
                              <td>
                                        2
                              </td>
                              <td>
                                        <NameInput
                                                  onChange={[Function]}
                                                  subject={
                                                            Object {
                                                                      "active": true,
                                                                      "advancedScore": 82,
                                                                      "elementaryScore": 100,
                                                                      "id": "0",
                                                                      "maxScore": Object {
                                                                        "formula": "A/B",
                                                                        "value": 182,
                                                                      },
                                                                      "name": "Matematyka",
                                                                    }
                                                  }
                                                  subjectsType="primary"
                                        />
                              </td>
                              <td>
                                        <ScoreRangeInput
                                                  examType="elementary"
                                                  onChange={[Function]}
                                                  subject={
                                                            Object {
                                                                      "active": true,
                                                                      "advancedScore": 82,
                                                                      "elementaryScore": 100,
                                                                      "id": "0",
                                                                      "maxScore": Object {
                                                                        "formula": "A/B",
                                                                        "value": 182,
                                                                      },
                                                                      "name": "Matematyka",
                                                                    }
                                                  }
                                                  subjectsType="primary"
                                        />
                                        <ScoreTextInput
                                                  examType="elementary"
                                                  onChange={[Function]}
                                                  subject={
                                                            Object {
                                                                      "active": true,
                                                                      "advancedScore": 82,
                                                                      "elementaryScore": 100,
                                                                      "id": "0",
                                                                      "maxScore": Object {
                                                                        "formula": "A/B",
                                                                        "value": 182,
                                                                      },
                                                                      "name": "Matematyka",
                                                                    }
                                                  }
                                                  subjectsType="primary"
                                        />
                                        %
                              </td>
                              <td>
                                        <ScoreRangeInput
                                                  examType="advanced"
                                                  onChange={[Function]}
                                                  subject={
                                                            Object {
                                                                      "active": true,
                                                                      "advancedScore": 82,
                                                                      "elementaryScore": 100,
                                                                      "id": "0",
                                                                      "maxScore": Object {
                                                                        "formula": "A/B",
                                                                        "value": 182,
                                                                      },
                                                                      "name": "Matematyka",
                                                                    }
                                                  }
                                                  subjectsType="primary"
                                        />
                                        <ScoreTextInput
                                                  examType="advanced"
                                                  onChange={[Function]}
                                                  subject={
                                                            Object {
                                                                      "active": true,
                                                                      "advancedScore": 82,
                                                                      "elementaryScore": 100,
                                                                      "id": "0",
                                                                      "maxScore": Object {
                                                                        "formula": "A/B",
                                                                        "value": 182,
                                                                      },
                                                                      "name": "Matematyka",
                                                                    }
                                                  }
                                                  subjectsType="primary"
                                        />
                                        %
                              </td>
                              <td>
                                        <ToggleSubject
                                                  active={true}
                                                  id="0"
                                                  onToggleClick={[Function]}
                                                  subjectsType="primary"
                                        />
                              </td>
                              <td>
                                        <DeleteSubject
                                                  id="0"
                                                  onDelete={[Function]}
                                                  subjectsType="primary"
                                        />
                              </td>
                    </tr>
          </tbody>
</table>,
        "_debugID": 2,
        "_renderedOutput": <table
          className="table mb-0"
>
          <tbody>
                    <tr>
                              <th>
                                        ID
                              </th>
                              <th>
                                        Przedmiot
                              </th>
                              <th>
                                        Wynik z matury podstawowej
                              </th>
                              <th>
                                        Wynik z matury rozszerzonej
                              </th>
                              <th />
                              <th />
                    </tr>
                    <tr>
                              <td>
                                        2
                              </td>
                              <td>
                                        <NameInput
                                                  onChange={[Function]}
                                                  subject={
                                                            Object {
                                                                      "active": true,
                                                                      "advancedScore": 82,
                                                                      "elementaryScore": 100,
                                                                      "id": "0",
                                                                      "maxScore": Object {
                                                                        "formula": "A/B",
                                                                        "value": 182,
                                                                      },
                                                                      "name": "Matematyka",
                                                                    }
                                                  }
                                                  subjectsType="primary"
                                        />
                              </td>
                              <td>
                                        <ScoreRangeInput
                                                  examType="elementary"
                                                  onChange={[Function]}
                                                  subject={
                                                            Object {
                                                                      "active": true,
                                                                      "advancedScore": 82,
                                                                      "elementaryScore": 100,
                                                                      "id": "0",
                                                                      "maxScore": Object {
                                                                        "formula": "A/B",
                                                                        "value": 182,
                                                                      },
                                                                      "name": "Matematyka",
                                                                    }
                                                  }
                                                  subjectsType="primary"
                                        />
                                        <ScoreTextInput
                                                  examType="elementary"
                                                  onChange={[Function]}
                                                  subject={
                                                            Object {
                                                                      "active": true,
                                                                      "advancedScore": 82,
                                                                      "elementaryScore": 100,
                                                                      "id": "0",
                                                                      "maxScore": Object {
                                                                        "formula": "A/B",
                                                                        "value": 182,
                                                                      },
                                                                      "name": "Matematyka",
                                                                    }
                                                  }
                                                  subjectsType="primary"
                                        />
                                        %
                              </td>
                              <td>
                                        <ScoreRangeInput
                                                  examType="advanced"
                                                  onChange={[Function]}
                                                  subject={
                                                            Object {
                                                                      "active": true,
                                                                      "advancedScore": 82,
                                                                      "elementaryScore": 100,
                                                                      "id": "0",
                                                                      "maxScore": Object {
                                                                        "formula": "A/B",
                                                                        "value": 182,
                                                                      },
                                                                      "name": "Matematyka",
                                                                    }
                                                  }
                                                  subjectsType="primary"
                                        />
                                        <ScoreTextInput
                                                  examType="advanced"
                                                  onChange={[Function]}
                                                  subject={
                                                            Object {
                                                                      "active": true,
                                                                      "advancedScore": 82,
                                                                      "elementaryScore": 100,
                                                                      "id": "0",
                                                                      "maxScore": Object {
                                                                        "formula": "A/B",
                                                                        "value": 182,
                                                                      },
                                                                      "name": "Matematyka",
                                                                    }
                                                  }
                                                  subjectsType="primary"
                                        />
                                        %
                              </td>
                              <td>
                                        <ToggleSubject
                                                  active={true}
                                                  id="0"
                                                  onToggleClick={[Function]}
                                                  subjectsType="primary"
                                        />
                              </td>
                              <td>
                                        <DeleteSubject
                                                  id="0"
                                                  onDelete={[Function]}
                                                  subjectsType="primary"
                                        />
                              </td>
                    </tr>
          </tbody>
</table>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Subject
    onChange={[Function]}
    onDelete={[Function]}
    onToggleClick={[Function]}
    subject={
        Object {
            "active": true,
            "advancedScore": 82,
            "elementaryScore": 100,
            "id": "0",
            "maxScore": Object {
              "formula": "A/B",
              "value": 182,
            },
            "name": "Matematyka",
          }
    }
    subjectNumber={2}
    subjectsType="primary"
/>,
}
`;
